Adjust `NcDatetimePicker` styling

raimund-schluessler opened the pull request:
This PR adjust the `NcDatetimePicker` styling to bring it inline with the other picker and input styles. It also replaces the icons used with material design icons.

| Before | After |
|-|-|
| ![Screenshot 2023-02-23 at 13-52-58 Nextcloud Vue Style Guide](https://user-images.githubusercontent.com/2496460/220912141-b082aa46-397e-44bf-850a-a091c6db6de7.png) | ![Screenshot 2023-02-23 at 13-53-23 Nextcloud Vue Style Guide](https://user-images.githubusercontent.com/2496460/220912167-1e88b73d-5c3c-44c4-a47c-111f6e035b01.png) |
| <img width="265" alt="ActionInputBefore" src="https://user-images.githubusercontent.com/2496460/220912201-5b232a6a-3cfa-492d-b11d-29e7932c53d1.png"> | <img width="266" alt="ActionInputAfter" src="https://user-images.githubusercontent.com/2496460/220946493-01ca7284-e037-4cd7-933a-0b7cac62145d.png"> |

Closes #3809.

susnux reviewed (approved):
The border looks darker than the others, but I think this is because of #3814

raimund-schluessler commented:
> The border looks darker than the others, but I think this is because of #3814

Yes, exactly. I rebased the branch with #3814 in and updated the screenshot. It looks the same now.

jancborchardt commented:
@raimund-schluessler just a super detail → when used in the action menu, is the text of the datepicker a bit too far down vertically inside the input? And the "calendar" icon on the right of the input is also not the same position for both of the actions for some reason?
![image](https://user-images.githubusercontent.com/925062/221537535-ab125626-c786-4d5e-98e3-f04564183be5.png)

raimund-schluessler commented:
> when used in the action menu, is the text of the datepicker a bit too far down vertically inside the input?

Might be, but I think this comes from the datepicker styling. See the standalone datepicker:
![Screenshot 2023-02-27 at 11-29-01 Nextcloud Vue Style Guide](https://user-images.githubusercontent.com/2496460/221539559-27b64f57-9a6f-4b7b-9af1-b191056b00dd.png)
But to be honest, I find it difficult to say what's the correct position, so I would leave it up to you to fix it 🙈 

> And the "calendar" icon on the right of the input is also not the same position for both of the actions for some reason?

Yes, it is. And until now, I thought we had no chance to fix this, as it's an icon coming from the browser. But I just noticed it reacts to the padding of the input field. So I will prepare a PR to fix it.

raimund-schluessler commented:
@jancborchardt See https://github.com/nextcloud/nextcloud-vue/pull/3826.
